### DISCUSSION TITLE
How can I obtain a token with the Authentication Code flow in a terminal?

### DISCUSSION THREAD

**Discussion by madorian** [keycloak/keycloak#22158] [Q&A]
Is there any code to complete an Authentication Code flow entirely in a terminal?

I've found snippets that launch a graphical browser from a terminal and instantiates a local server to get the token from the callback, but I don't want to see a graphical browser. 

Any pointers as to how this can be done?;)

**Comment by embesozzi** [comment]
Hi, @madorian.
There are different OAuth 2.0 flows for doing the federation process. For instance, for devices you can use the [Device Flow]( https://datatracker.ietf.org/doc/html/draft-ietf-oauth-device-flow-15).

**Reply by madorian** [reply]
The providers I intend to use doesn't use Device Authorization. I just want to automate this process, cause it's easier to automate it, than to use the mouse to clicky clicky all the time to get a token.

**Reply by embesozzi** [reply]
In the case of the Authentication Code flow, you need to open a browser, webview, etc. For testing purposes, scenarios can be automated using frameworks like Selenium to handle the login process.
